ember-resources/src/util/function.ts

```diff
@@ -138,36 +139,76 @@ export function trackedFunction<Return>(...passedArgs: UseFunctionArgs<Return>)
  * State container that represents the asynchrony of a `trackedFunction`
  */
 export class State<Value> {
-  @tracked isResolved = false;
```

Comment:
can we have isResolved and isSuccessful together? -- we can't actually get rid of isResolved (or any other property), as that is a breaking change.

Reply:
What do you mean "have them together"?

They mean different things. Originally, whent he promise was rejected, `isResolved` would become `true` ([source](https://github.com/NullVoxPopuli/ember-resources/blob/824404911d/ember-resources/src/util/function.ts#L193-L195))).

Since it was very counter intuitive to have `isResolved` as `true` when the promise was in fact not resolved, I chose to remove this flag and replace with unambiguous ones: `isSuccessful` and `isFinished`, borrowing from Ember Concurrency.

I believe that this is reason is very worthy of a breaking change and a major version bump.

If you disagree, I can make `isResolved` a getter alias. But note that it will have to be an alias to `isFinished` and not to `isResolved` as you say in the previous comment.

Reply:
> What do you mean "have them together"?

Like this:

> I can make isResolved a getter alias. But note that it will have to be an alias to isFinished and not to isResolved as you say in the previous comment.

sounds good to me!

>  If you disagree, 

yeah -- mostly wanting to avoid a major bump atm

> I believe that this reason is very worthy of a breaking change and a major version bump.

unfortunately, `trackedFunction` is versioned with `ember-resources`, and there have been no significant enough changes to warrant a major bump. There are some things I'd like to explore, and we can plan out what we want for a major, but idk how soon we'd want to release that.
(things I've been wanting to see implemented soon: registerable usables, a resource blueprint, etc)

Reply:
>> I can make isResolved a getter alias. But note that it will have to be an alias to isFinished and not to isResolved as you say in the previous comment.

> sounds good to me!

Will do.

***

> unfortunately, trackedFunction is versioned with ember-resources, and there have been no significant enough changes to warrant a major bump.

Please excuse me for starting a holy war, but I believe that SemVer goal is to capture breaking changes, however small, and NOT to capture the significance of changes. A spiking major version number is an indication of proper SemVer usage.

I will add a alias `isResolved`, but I object: this property will become `true` when the promise is not resolved. This should be fixed, and there is no point in holding to a major version number.